packages/engine/lib/orchestrator/src/experiment.rs

```diff
@@ -95,6 +95,25 @@ pub struct ExperimentConfig {
     clap(global = true, short = 'w', long, default_value_t = num_cpus::get(), validator = at_least_one, env = "HASH_WORKERS")
     )]
     pub num_workers: usize,
+
+    #[cfg_attr(feature = "clap", clap(global = true, long, default_value_t = 0))]
+    /// Initial size of the V8 heap.
+    ///
+    /// Setting this value allows to avoid garbage collection while the heap is small enough.
+    pub v8_initial_heap_constraint: usize,
+
+    #[cfg_attr(
+        feature = "clap",
+        clap(global = true, long, default_value_t = 4_000_000_000)
+    )]
+    /// Max size of the V8 heap.
+    ///
+    /// V8 will run a series of garbage collection when the heap size gets close to this limit.
+    ///
+    /// If garbage collection can't get the heap smaller than this limit then it crashes.
+    ///
+    /// Defaults to 4GB.
+    pub v8_max_heap_constraint: usize,
```

Comment:
Same as above, maybe it makes sense to simply use `Option<usize>` and fall back to the default heap size provided by v8?

Reply:
One of the main part of the task was to set this value because the default is not big enough.
If the default was ok then I don't understand why this task was needed at all.

Reply:
V8 chooses a sensible default for the max-heap constraint: It's using physical and logical memory sizes. The task does not say, that we should set a fixed limit, we want to make it configurable. If a user does not provide a value, why can't we fall back to the default provided by v8?

Reply:
V8 chooses a sensible default for the max-heap constraint: It's using physical and logical memory sizes. The task does not say, that we should set a fixed limit, we want to make it configurable. If a user does not provide a value, why can't we fall back to the default provided by v8?

Reply:
Turns out we were both wrong. There is a method using the available RAM but it seems not to be used by default.

Looking at the [source code](https://github.com/v8/v8/blob/faaf6818d289b874699c40f925b07ed9da1d8f2b/src/heap/heap.cc#L5108):

- `max_semi_space_size = 8 * (kSystemPointerSize / 4) * MB` (clamped by `kMinSemiSpaceSize`)
- `max_old_generation_size = 700ul * (kSystemPointerSize / 4) * MB` (Somewhat clamped by page-size)
- **`max_global_memory_size = max_old_generation_size * 2`**

Which results in 3 GB by default. 

> - We have this limit set to some value on the cloud

Yes, the limits in cloud can be found at the `ffi` file by searching `ConfigureDefaultsFromHeapSize` (or use [cs.github.com](https://cs.github.com)).

> - The default is too small and is not configurable

Yes, the default is too small for big-scaled simulations. For small simulations (like everything in our `examples/` folder) it's fine.

> - We have to come up with a good default and possibly make it configurable

As all simulations we are currently testing are working, we agreed on using v8's default value and simply making it configurable. As long as we don't have concrete numbers, we can't really set a sensible default. This is why I asked you to just use `Option<usize>` instead of `usize` for the other parameter. `0` means nothing and is even more confusing than having no information because this is nowhere stated in the official documentation (at least I wasn't able to find that in 2 hours). This means, that this is an implementation detail of v8.

Reply:
Use `Option` in https://github.com/hashintel/hash/pull/446/commits/39550930d1416cf96fc927fb51721b2c0c7c0f77